src/content/docs/en/guides/imports.mdx

```diff
@@ -221,6 +232,96 @@ const components = await Astro.glob('../components/*.astro');
 ))}
 ```
 
+### Supported Values
+
+Vite's `import.meta.glob()` function only supports static string literals. It does not support dynamic variables and string interpolation.
+
+A common workaround is to instead import a larger set of files that includes all the files you need, then filter them:
+
+```astro {6-7}
+---
+// src/components/featured.astro
+const { postSlug } = Astro.props;
+const pathToMyFeaturedPost = `src/pages/blog/${postSlug}.md`;
+
+const posts = await Astro.glob('../pages/blog/*.md');
+const myFeaturedPost = posts.find(post => post.file.includes(pathToMyFeaturedPost));
+---
```

Comment:
From the context, I think we should use `import.meta.glob` instead of `Astro.glob` in this example. However, I'm not sure about the right way to use it.

I created a minimal example locally without adding Typescript (from the prompt) and it seems that, at least in VSCode, we still need to provide a type to `import.meta.glob` to avoid errors (e.g. `Property 'file' does not exist on type 'unknown'.ts(2339)`). I think this is because a `tsconfig.json` file is created in all cases.

So the example should be something like:

```suggestion
---
// src/components/featured.astro
interface Frontmatter {
  title: string;
  description?: string;
}

const { postSlug } = Astro.props;
const pathToMyFeaturedPost = `src/pages/blog/${postSlug}.md`;

const posts = Object.values(
  import.meta.glob<MarkdownInstance<Frontmatter>>("../pages/blog/*.md", { eager: true })
);
const myFeaturedPost = posts.find(post => post.file.includes(pathToMyFeaturedPost));
---
```

We could remove the type to simplify the example but my concern is that some users may copy/paste the code without understanding why their editor is complaining.

However using that updated example makes the next section (`Import type utilities`) a bit awkward since we are more or less repeating the `Markdown files` section example.

So I see three options:
* rewrite the example without Typescript type/interface but add the extra checks for `file`, `url` and `frontmatter.title` (but that might make the example a bit too complex).
* rewrite the example without Typescript type/interface and add a warning to inform users that the example is not complete and that without a Typescript type they'll need additional checks to make sure `file` and the other properties exist to avoid errors in their editor.
* reorganize the section to only use Typescript types and maybe add a note at the end to explain that `import.meta.glob` can be used without Typescript types but by doing it this way, the user will have to add additional checks to verify the shape.

I'm not sure which is the best choice here.

Reply:
I think it's fine to not use TypeScript in this example, as many of our examples currently don't. This example is more to illustrate the point of over searching and filtering, and it's not likely they'll copy this code because the variables won't be of much use to them.

However, I think I will probably remove it anyway for Reasons:tm: and I believe we show something similar in content collections about filtering a full collection fetch to only get *some* results.

I'm not sure how the `Astro.glob` missed the find and replace here though! I thought I had caught them all!

Reply:
I committed the quick fix to remove Astro.glob, but ideally I'd also *like* for this example to maybe be fetching some images (like maybe using our dynamic imported images recipe?) and filtering for only "cat photos" or something?

Would you be interested in/willing to whip up an example here that isn't about fetching markdown files, but maybe images @ArmandPhilippot ? Again, I don't think TypeScript is necessary *here* to just show the idea of fetching more, then filtering.

Reply:
@sarah11918 I can give it a try! 🫡  Since there is already an existing base, it will help me (never used `import.meta.glob` before)

Reply:
Thank you! Again, I'm only looking really for a line showing an import, then a line that makes a new array of just a subset somehow. It doesn't need to be fancy, just illustrative of what we mean when we say "import more than you'll need, and narrow it down"  :smile: